components/experimental/src/dimension/provider/currency_patterns.rs

```diff
@@ -17,11 +17,58 @@ use icu_provider::prelude::*;
 #[icu_provider::data_struct(marker(CurrencyPatternsDataV1Marker, "currency/patterns@1"))]
 #[derive(Debug, Clone, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize))]
-#[cfg_attr(feature = "datagen", derive(serde::Serialize, databake::Bake))]
-#[cfg_attr(feature = "datagen", databake(path = icu_experimental::dimension::provider::currency_patterns))]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
 #[yoke(prove_covariance_manually)]
 pub struct CurrencyPatternsDataV1<'data> {
     /// Contains the unit patterns for a currency based on plural rules.
     #[cfg_attr(feature = "serde", serde(borrow))]
     pub patterns: PluralPatterns<'data, DoublePlaceholder>,
 }
+
+impl<'data> CurrencyPatternsDataV1<'data> {
+    /// Construct an instance directly from a byte slice.
+    ///
+    /// # Safety
+    ///
+    /// The bytes must have been returned by [`Self::as_byte_slice`].
+    pub const unsafe fn from_byte_slice_unchecked(bytes: &'data [u8]) -> Self {
+        Self {
+            patterns: PluralPatterns {
+                strings: icu_plurals::provider::PluralElementsPackedCow {
+                    elements: alloc::borrow::Cow::Borrowed(
+                        icu_plurals::provider::PluralElementsPackedULE::from_byte_slice_unchecked(
+                            bytes,
+                        ),
+                    ),
+                },
+                _phantom: core::marker::PhantomData,
+            },
+        }
+    }
+
+    /// Returns this instance as a byte slice.
+    #[cfg(feature = "datagen")]
+    pub fn as_byte_slice(&self) -> &[u8] {
```

Comment:
nit: must this be pub?

Reply:
Done. I merged it into the Bake impl and updated the comments accordingly